### PR TITLE
[NOMERGE] New package: adobe-flash-package

### PR DESCRIPTION
**Pull request by masato9000**
Package containing build scripts for current version of Adobe flash, for packaging in xbps.
Covers both NPAPI and PPAPI.

Not as brain-in-neutral easy as having the INSTALL routine automatically download and copy files (chromium-pepper-flash) but safer as you don't have any files unmanaged by xbps.  After installation of this package, the user/admin runs a command to download/package and optionally install the locally-created binary package.

Two main purposes:
* Safe for the repos (no distribution of flash binaries), so flash can show up on package search.
* All non-temp files managed by XBPS, rather than INSTALL/REMOVE scripts.

Master package contains appropriate nondistribution notification, and created packages contain license/nonredistribution warnings as well.

### CODE REVIEW

File: srcpkgs/adobe-flash-package/files/pepper-flash-package [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	--pkgver "${ppapipkg}-${VERSION}_1" \
+	--conflicts "${conflicts_ppapi}" \
+	--compression xz \
+	--quiet ${DEST_DIR}
+
+rm -r ${BUILD_DIR}
+
+if [ ${_doinstall} -eq 1 ]; then
+	echo y | xdowngrade ${ppapipkg}-${VERSION}_1.${ARCH}.xbps
+fi
```
Comment (Gottox) [inline srcpkgs/adobe-flash-package/files/pepper-flash-package]: Same here. These two scripts may can be merged into one, I guess.

File: srcpkgs/adobe-flash-package/files/pepper-flash.sh.in [hunk 2]
```diff
@@ -0,0 +1 @@
+export CHROME_FLAGS="--ppapi-flash-path=/usr/lib/%PP_PKG%/libpepflashplayer.so --ppapi-flash-version=%VERSION%"
```
Comment (Gottox) [inline srcpkgs/adobe-flash-package/files/pepper-flash.sh.in]: add newline at the end

File: srcpkgs/adobe-flash-package/files/adobe-flash-package [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+	--pkgver "${npapipkg}-${VERSION}_1" \
+	--conflicts "${conflicts_npapi}" \
+	--compression xz \
+	--quiet ${DEST_DIR}
+
+rm -r ${BUILD_DIR}
+
+if [ ${_doinstall} -eq 1 ]; then
+	echo y | xdowngrade ${npapipkg}-${VERSION}_1.${ARCH}.xbps
+fi
```
Comment (Gottox) [inline srcpkgs/adobe-flash-package/files/adobe-flash-package]: rather then building a package and autoinstalling it, add the files in the INSTALL hook and remove them in the REMOVE hook.

File: srcpkgs/adobe-flash-package/files/pepper-flash-package (hunk 1 above)
Comment (thypon) [inline srcpkgs/adobe-flash-package/files/pepper-flash-package]: Tracking these files with xbps is not a bad idea btw, instead of removing with "rm -rf"

Comment (Gottox) [inline srcpkgs/adobe-flash-package/files/pepper-flash-package]: It's a lot of pointless file extracting, packaging, again extracting. So for now I'd rather go with the INSTALL/REMOVE script approach.

There are some ideas floating around to let install scripts add files to packages but it isn't anything that will come soon.

File: srcpkgs/adobe-flash-package/files/pepper-flash.sh.in (hunk 2 above)
Comment (masato9000) [inline srcpkgs/adobe-flash-package/files/pepper-flash.sh.in]: ok, this one's done

File: srcpkgs/adobe-flash-package/files/pepper-flash-package (hunk 1 above)
Comment (masato9000) [inline srcpkgs/adobe-flash-package/files/pepper-flash-package]: I actually had two installer packages (one for each flavor) using that method that I had initially intended to commit, but some have expressed displeasure at that method (chromium-pepper-flash) as it was a dirty hack.

So I came up with this one, which tries to be a less dirty (and less dangerous) hack.  No scripts trying to remove files from /usr directory.

This way is a bit more roundabout, but again, safer.  I do like the result, but I still have the other two packages.  So, I'm open.

File: srcpkgs/adobe-flash-package/files/adobe-flash-package (hunk 3 above)
Comment (masato9000) [inline srcpkgs/adobe-flash-package/files/adobe-flash-package]: Combining the two has crossed my mind, and just using flags to determine which routine to use, though I'm not sure if there are any real gains.  I held off on the idea as I wanted see how acceptable the whole idea was.  (see other comments^^^)

**Comment by Vaelatern** [comment]
@Gottox ?

File: srcpkgs/adobe-flash-package/files/pepper-flash-package (hunk 1 above)
Comment (masato9000) [inline srcpkgs/adobe-flash-package/files/pepper-flash-package]: @Gottox 
additionally, if neither method seems very good, then we can always go back to the restricted template; the one we have for npapi is probably in need of an update (I haven't checked), and we can whip up the same type of thing for pepper.

It means flash wouldn't show up on a package search, but I haven't seen many questions on the forum about flash (I don't know how it is on the irc channel).  So perhaps users have accepted the idea of getting it from adobe, like you do with a certain other operating system...  And in that case, the restricted template may be enough for users who would rather have the files managed by xbps than copy/replace them manually.

**Comment by TheDoctorsLife** [comment]
@masato9000 The package is out of date, the new version is 221. Here is a patch for that version: http://sprunge.us/gBhR

**Comment by masato9000** [comment]
Changing this to NOMERGE; I'm finishing up a new script that does what this package wants to do but with some differences.
It's a single script (still packages each flavor separately because I still like that better).  It does new-version checks (can be put in cron; can send notifications to console/terminal or libnotify when new versions are available).

That'll be in a new repo on my github account once I finish up.  Since it's just 1 script, and not very big, I don't think I'll try to get it into a Void package (it's just easier to just drop it in /usr/local/bin and modify as you like).

Leaving this open for now, in case anybody that was watching it has any suggestions (otherwise, it seems to me that the topic isn't terribly interesting to many voiders)

**Comment by masato9000** [comment]
I have the unified script on [gist](https://gist.github.com/masato9000/be7f217a11b657e458f07e9801f146ee).

**Pull request closed without merging**